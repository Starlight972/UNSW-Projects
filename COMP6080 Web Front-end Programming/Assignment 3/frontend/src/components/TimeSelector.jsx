import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

// MUI components
import InputLabel from '@mui/material/InputLabel';
import FormControl from '@mui/material/FormControl';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';

const TimeSerlector = ({ setTime, time }) => {
  return (
    <Wrapper>
      <InputLabel>Time Limit</InputLabel>
      <FormControl fullWidth>
      <Select
        style={{ background: 'white' }}
        value={time}
        onChange={(e) => setTime(e.target.value)}
      >
        <MenuItem value= {5} >5 s</MenuItem>
        <MenuItem value= {10}>10 s</MenuItem>
        <MenuItem value= {20}>20 s</MenuItem>
        <MenuItem value= {30}>30 s</MenuItem>
        <MenuItem value= {60}>1 min</MenuItem>
        <MenuItem value= {90}>1 min 30 s</MenuItem>
        <MenuItem value= {120}>2 min</MenuItem>
        <MenuItem value= {240}>4 min</MenuItem>
      </Select>
      </FormControl>
    </Wrapper>
  )
}

export default TimeSerlector;

TimeSerlector.propTypes = {
  time: PropTypes.any,
  setTime: PropTypes.any
}

const Wrapper = styled.div`
  width: 45%;
  margin: 5px;
  minWidth: 165px;
`;
