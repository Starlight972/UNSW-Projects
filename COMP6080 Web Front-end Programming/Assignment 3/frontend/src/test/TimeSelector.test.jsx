import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';

// MUI components
import InputLabel from '@mui/material/InputLabel';
import FormControl from '@mui/material/FormControl';
import Select from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';

// own component
import TimeSelector from '../components/TimeSelector';

describe('<TimeSelector />', () => {
  const noop = () => {};

  it('renders without crashing', () => {
    const component = shallow(<TimeSelector setTime={noop} />);
    console.log(component);
  })

  it('matches snapshot', () => {
    const component = shallow(<TimeSelector setTime={noop} />);
    expect(component).toMatchSnapshot();
  });

  it('should has elements correct', () => {
    const component = shallow(<TimeSelector setTime={noop} />);
    expect(component.find(InputLabel)).toHaveLength(1);
    expect(component.find(InputLabel).text()).toEqual('Time Limit');
    expect(component.find(FormControl)).toHaveLength(1);
    expect(component.find(Select)).toHaveLength(1);
    expect(component.find(MenuItem)).toHaveLength(8);
  })

  it('uses fallback time if none provided', () => {
    const component = shallow(<TimeSelector setTime={noop} />);
    expect(component.find(Select).props().value).toBe(undefined);
  })

  it('uses fallback time if provided', () => {
    const component = shallow(<TimeSelector setTime={noop} time={5} />);
    expect(component.find(Select).props().value).toBe(5);
  })

  it('triggers onChange event handler when clicked', () => {
    const onChange = jest.fn();
    const component = shallow(<TimeSelector setTime={onChange} />);
    const select = component.find(Select);
    select.simulate('change', { target: { value: 5 } });
    expect(onChange).toHaveBeenCalledTimes(1);
  })
})
