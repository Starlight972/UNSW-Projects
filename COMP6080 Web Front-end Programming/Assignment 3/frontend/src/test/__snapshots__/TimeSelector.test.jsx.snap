// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeSelector /> matches snapshot 1`] = `ShallowWrapper {}`;
